src/engraving/libmscore/excerpt.cpp

```diff
@@ -1198,7 +1198,7 @@ void Excerpt::cloneStaff(Staff* srcStaff, Staff* dstStaff)
                 dstTrack2 = dstStaffIdx * VOICES + (s->track2() % VOICES);
             }
         }
-        if (dstTrack == mu::nidx) {
+        if (dstTrack == mu::nidx || staffIdx == srcStaffIdx ) {
             continue;
         }
         cloneSpanner(s, score, dstTrack, dstTrack2);
```

Comment:
cloneSpanner(s, score, dstTrack, dstTrack2) clones all spanners not only srcStaff.